Client/Frontend/Browser/Tabs/TabTrayViewController.swift

```diff
@@ -332,7 +356,12 @@ extension TabTrayViewController {
     }
 
     @objc func didTapSyncTabs(_ sender: UIBarButtonItem) {
+        syncTabButton.isEnabled = false
         viewModel.didTapSyncTabs(sender)
+        
+        DispatchQueue.main.async { [weak self] in
+            self?.syncTabButton.customView = self?.syncLoadingView
```

Comment:
one last comment: We should keep be updating UI elements like the button title in one place, if possible. We should just call the `updateTitle` function here, and it should do update the title based on whether it's done syncing or starting the sync. That way, when we're updating design/code later, we only have to look in one place.